js/src/admin.js

```diff
@@ -2,6 +2,67 @@
  * @package Polylang
  */
 
+window.pll = window.pll || {};
+
+/**
+ * Calls the passed callable when the DOM is ready.
+ *
+ * @param {callable} fn - Function to call when the DOM is ready.
+ */
+window.pll.dom_ready = ( fn ) => {
+	if ( document.readyState !== 'loading' ) {
+		fn();
+	} else {
+		document.addEventListener( 'DOMContentLoaded', fn );
+	}
+};
+
+/**
+ * Settings: manages AJAX calls attached to `.pll-ajax-button` buttons.
+ */
+window.pll.settings_ajax_button = () => {
+	document.querySelectorAll( '.pll-ajax-button' ).forEach( ( el ) => {
+		el.addEventListener( 'click', ( event ) => {
+			const button    = event.target;
+			const action    = button.getAttribute( 'data-action' );
+			const nonce     = button.getAttribute( 'data-nonce' );
+			const row       = button.closest( 'tr' );
+			const errorElms = row.querySelectorAll( '.pll-error-message-text' );
+
+			if ( ! action || ! nonce || ! row || ! errorElms.length ) {
+				return;
+			}
+
+			const sep = ( ajaxurl.indexOf( '?' ) !== -1 ) ? '&' : '?';
```

Comment:
See if we couldn't also depend on `wp.url` package? 🤔
There are also functions available in this package like `addQueryArgs` which already manage the query string separator.

Reply:
Done in [6ceaf75](https://github.com/polylang/polylang/pull/1420/commits/6ceaf753e97aba3322e52be876406ff58e3d5424).